### ISSUE TITLE
[Screen Reader -  Bot Framework Web Chat- Card Sports] Descriptive information about the 'image' link controls is not announced which appears after sending 'card sports' text

### ISSUE THREAD

**Issue by mishraadi** [microsoft/BotFramework-WebChat#4025]
User Experience:
If descriptive information about the 'image' link controls is not announced which appears after sending 'card sports' text, the screen reader dependent user will not get proper descriptive information about the link control. The user will face issue in navigating to the control and utilizing it's functionality.

Note: Login Credentials must not be provided within the bug.

Test Environment:
OS Version: 21H1
OS Build: 19043.1110
Browser: Microsoft Edge Version 94.0.972.2 (Official build) dev (64-bit)
Screen Reader: Narrator, JAWS, NVDA

Repro steps:

1. Open URL provided by product team.
2. Web chat bundle will open.
3. Navigate on the web chat bot and select microphone icon.
4. Speak "help" keyword in Type your message box, then select send button.
5. Cards related to the keyword will appear, navigate on the cards.
6. Verify the issue.

Actual Result
Descriptive information about the 'image' link controls is not announced which appears after sending 'card sports' text. Only 'Image' and control type as link is announced when focus is navigated on the control.

Expected Result
Descriptive information about the 'image' link controls should be announced which appears after sending 'card sports' text. When the focus reaches on the control, proper role and image description should be announced for each and every control.

Note: 
This issue exists with NVDA and JAWS also.
This issue exists with similar controls throughout the application.

WCAG Reference
WCAG 1.3.1 – Info and Relationships

Reference Links
Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.

**Comment by mishraadi** [comment]
#A11yMAS;#A11ySev2;#HCL;#MAS1.3.1;#NonBenchmark;#Accessibility;#Win10-Edge(Chromium);#E2E-HCL-BotFrameworkWebChat-Aug21;#AILimited;

**Comment by compulim** [comment]
This is an Adaptive Cards issue.

Could you file it to them at https://github.com/microsoft/adaptivecards?

FYI, I have attached the content of the card here.

[BingSports.txt](https://github.com/microsoft/BotFramework-WebChat/files/6984444/BingSports.txt)

**Comment by compulim** [comment]
I have attached the DOM node for your reference.

![image](https://user-images.githubusercontent.com/1622400/130521049-4eeeb0d0-2ad6-40ff-8612-d45e41b3224c.png)

> Descriptive information about the 'image' link controls should be announced which appears after sending 'card sports' text. When the focus reaches on the control, proper role and image description should be announced for each and every control.

On the DOM node, I am seeing both `alt` and `role` set. And it seems to fulfill the requirement ("proper role and image description").

As we do not own the Adaptive Cards control, if this is an issue, please file it to Adaptive Cards team.

**Comment by compulim** [comment]
Studied a bit more. It seems it is an Adaptive Card problem. They should not convolute the link image with the `<img role="link">`.

When we try to reproduce the DOM tree as rendered by Adaptive Cards library. It becomes:

```html
<img role="link" alt="Logo" aria-label="Action" />
```

When combining link and image as a single element, the screen reader will NOT narrate the `alt` text.

The following is our pure HTML repro. Given 2 image links below: Panthers and Seahawks. The Panthers logo is `<a><img /></a>`. The Seahawks logo is `<img role="link">`.

```html
<div>
  <a href="https://bing.com/" aria-label="Open URL action">
    <img
      alt="Panthers logo"
      src="https://webchat-mockbot.azurewebsites.net/public/assets/sports-panthers.png"
    />
  </a>
</div>

<div>
  <img
    alt="Seahawks logo"
    aria-label="Open URL action"
    role="link"
    src="https://webchat-mockbot.azurewebsites.net/public/assets/sports-seahawks.png"
    tabindex="0"
  />
</div>
```

The Panthers logo, when navigating using screen reader, we will hear:

- "Link image open URL action", when <kbd>CAPS</kbd> + <kbd>DOWN</kbd> is pressed, it will be followed by
- "Panthers logo"

For the Seahawks logo, we will hear:

- "Link image open URL action"
-  After <kbd>CAPS</kbd> + <kbd>DOWN</kbd> is pressed, it will NOT narrate "Seahawks logo"

**Comment by compulim** [comment]
Filed a bug to Adaptive Cards repo, https://github.com/microsoft/AdaptiveCards/issues/6442.

**Comment by compulim** [comment]
Since the image label (a.k.a. alt text) is ignored, in the meanwhile, I will change the Adaptive Cards JSON to be more descriptive on the link label.

**Comment by v-shikch** [comment]
As verified this issue is fixed now. Here screen reader is announcing correct descriptive information about the image link. Hence closing the issue

**Comment by compulim** [comment]
Thanks @v-shikch and @mishraadi. Appreciated.
